client/blocks/login/login-form.jsx

```diff
@@ -257,6 +259,163 @@ export class LoginForm extends Component {
+					{ this.renderPrivateSiteNotice() }
+					<div className="login__form-userdata">
+						{ linkingSocialUser && (
+							<p>
+								{ this.props.translate(
+									'We found a WordPress.com account with the email address "%(email)s". ' +
+										'Log in to this account to connect it to your %(service)s profile, ' +
+										'or choose a different %(service)s profile.',
+									{
+										args: {
+											email: this.props.socialAccountLinkEmail,
+											service: capitalize( this.props.socialAccountLinkService ),
+										},
+									}
+								) }
+							</p>
+						) }
+
+						<label htmlFor="usernameOrEmail">
+							{ this.isPasswordView() ? (
+								<Button
+									borderless
+									className="login__form-change-username"
+									onClick={ this.resetView }
+								>
+									<Gridicon icon="arrow-left" size={ 18 } />
+
+									{ includes( this.state.usernameOrEmail, '@' )
+										? this.props.translate( 'Change Email Address' )
+										: this.props.translate( 'Change Username' ) }
+								</Button>
+							) : null }
+						</label>
+
+						<TextControl
+							label={ this.props.translate( 'Email Address or Username' ) }
+							disabled={ isFormDisabled || this.isPasswordView() }
+							id="usernameOrEmail"
+							name="usernameOrEmail"
+							value={ this.state.usernameOrEmail }
+							onChange={ value => {
+								this.props.formUpdate();
+								this.setState( {
+									usernameOrEmail: value,
+								} );
+							} }
+						/>
+
+						{ requestError && requestError.field === 'usernameOrEmail' && (
+							<FormInputValidation isError text={ requestError.message } />
+						) }
+
+						<div
+							className={ classNames( 'login__form-password', {
+								'is-hidden': this.isUsernameOrEmailView(),
+							} ) }
+						>
+							<TextControl
+								label={ this.props.translate( 'Password' ) }
+								disabled={ isFormDisabled }
+								id="password"
+								name="password"
+								type="password"
+								value={ this.state.password }
+								onChange={ value => {
+									this.props.formUpdate();
+									this.setState( {
+										password: value,
+									} );
+								} }
+							/>
+
+							{ requestError && requestError.field === 'password' && (
+								<FormInputValidation isError text={ requestError.message } />
+							) }
+						</div>
+					</div>
+
+					{ config.isEnabled( 'signup/social' ) && (
+						<p className="login__form-terms">
+							{ preventWidows(
+								this.props.translate(
+									// To make any changes to this copy please speak to the legal team
+									'By continuing with any of the options below, ' +
+										'you agree to our {{tosLink}}Terms of Service{{/tosLink}}.',
+									{
+										components: {
+											tosLink: (
+												<a
+													href={ localizeUrl( 'https://wordpress.com/tos/' ) }
+													target="_blank"
+													rel="noopener noreferrer"
+												/>
```

Comment:
Should you use `ExternalLink` component here?

Reply:
Done.